File: java/prime-factors/src/main/java/PrimeFactors.java
import java.util.ArrayList;
import java.util.List;

public class PrimeFactors {

  public static List<Long> getForNumber(long number) {
    List<Long> primes = new ArrayList();
    
    long factor = 2;
    long remainder = number;

    while (remainder != 1 && factor <= number) {
      if (remainder % factor == 0) {
        remainder /= factor;
        primes.add(factor);
      } else {
        factor += 1;
      }
    }

    return primes;
  }

}

File: python/prime-factors/prime_factors.py
def prime_factors(number):
    if number < 2:
        return []

    factor, factors = 2, []

    while number > 1:
        quotient, remainder = divmod(number, factor)

        if remainder == 0:
            factors.append(factor)
            number = quotient
        else:
            factor += 1

    return factors
